### PR TITLE
create a new page/new-project with submit and home button

### PR DESCRIPTION
**Pull request by AndryBasaraba**

### CODE REVIEW

File: components/TaskCard/Card.module.css
Comment (Kheirah) [inline components/TaskCard/Card.module.css:1]: `styled-components` instead of CSS.

File: components/Form/index.js [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        projectTitle: formData.title,
+        projectSize: formData.size,
+        projectDue: formData.dueBy,
+        tasks: formData.tasks.split(",").map((task) => task.trim()),
+      };
+
+      localStorage.setItem(
+        "projectsData",
+        JSON.stringify([...projectsData, newProject])
+      );
```
Comment (Kheirah) [inline components/Form/index.js:75]: This is something you could best put in a separate `useEffect` which reacts to a change in `projectData`.
After you call `setProjectData` the `useEffect` will notice that the state changed and run its callback.
```js
useEffect(() => {
    localStorage.setItem(
        "projectsData",
        JSON.stringify([...projectsData, newProject])
    );
}, [projectData]);
```

File: components/Form/index.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+            name="tasks"
+            value={formData.tasks}
+            onChange={handleInputChange}
+          />
+          {errors.tasks && <span className={styles.error}>{errors.tasks}</span>}
+        </label>
+        <button
+          type="submit"
+          className={styles.submit}
+          onClick={formSubmitHandler}
```
Comment (Kheirah) [inline components/Form/index.js:164]: The submit event is not triggered by the `onClick` of the submit button **but by the form.**
Add the `onSubmit` event to the form and move the `formSubmitHandler` up to the form.

File: components/Form/index.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    const validSizes = ["s", "m", "l", "xl"];
+    if (!validSizes.includes(formData.size.toLowerCase())) {
+      newErrors.size = "Size must be one of s, m, l, or xl";
+      isValid = false;
+    }
+
+    // Due by validation
+    const dueDate = new Date(formData.dueBy);
+    if (isNaN(dueDate) || dueDate < new Date()) {
+      newErrors.dueBy = "Due by must be a valid future date";
```
Comment (Kheirah) [inline components/Form/index.js:46]:
```suggestion
      newErrors.dueBy = "Pick a valid future date";
```

**Review by Kheirah — CHANGES_REQUESTED** [review]
Nice, but there's still quite a bit left to do.

One major change revolves around the use of raw css. Use `styled-components` instead.
Another rather small change is the submit event which needs to be catched by the form instead of triggered by the button.

**Comment by AndryBasaraba** [comment]
all comments have been corrected

File: components/Form/index.js (hunk 1 above)
Comment (Kheirah) [reply components/Form/index.js:87]: From the perspective of React `localStorage` is an external system. `localStorage` not part of React but of the browser. _Whenever you want to synchronize your app's state with an external system, you use `useEffect`._

In your case, you manage project data (your state) which you want to save to `localStorage` (the browser). Saving to the browser's `localStorage` is a considered a side-effect. You can [read more here](https://react.dev/learn/synchronizing-with-effects).

**Pull request merged**